packages/back-end/src/validators/saved-queries.ts

```diff
@@ -50,14 +50,59 @@ export type dimensionAxisConfiguration = z.infer<
   typeof dimensionAxisConfigurationValidator
 >;
 
-export const dataVizConfigValidator = z.object({
+const chartTypesThatRequireXAndYAxis = z.enum([
```

Comment:
nit: I feel like this is too restrictive of a design, since we may have multiple y-axes in the future. Is there not a way to just have the axes defined on the chart type directly instead of having a group for each type of chart depending on the number/type of axes it has?

Reply:
Awesome, yeah I like the future dev ex of this approach. I'll leave it to Adriel probably for the full review?